docs/batch_inference_with_mms.md

```diff
@@ -0,0 +1,335 @@
+        ...
+...        
+```
+This initialization logic binds the MXNet module with a batch-size that the model is configured with. For example, if the incoming requests are images with RGB layers and the input to the network is
+or size [3,224, 224], the batch adds an additional dimension to this data. If the batch size configured with `POST /models` API is 8, the input to the DL Framework becomes [8, 3, 224, 224]. 
+
+#### Preprocess logic
+Once a model is initialized, the DL framework (in our case MXNet) expects a `batch_size` number of requests. Hence, the preprocessing logic should make sure that no matter how many requests
+come into the handler, it always returns the `batch_size` number of NDArray elements to be used in the inference logic. Lets look at a sample `preprocess` logic below 
+
+```python
+import mxnet as mx
+
+class BatchingExample(object):
+...
+    def initialize(self, context):
+    ...
+...
+    def preprocess(self, request):
+        img_list = []
+        input_shape = self.signature['inputs'][0]['data_shape']
+        # We are assuming input shape is NCHW
+        [c, h, w] = input_shape[1:]
+        
+        for idx, data in enumerate(request):
+            img_arr = mx.image.imdecode(img, 1, True, None)
+            img = data.get("body")
+            img_arr = mx.image.imresize(img_arr, w, h, 2)
+            img_arr = mx.nd.transpose(img_arr, (2, 0, 1))
+            self._num_requests = idx + 1
+            img_list.append(img_arr)
+            
+            reqs = mx.nd.stack(*img_list)
+            reqs = reqs.as_in_context(self.mxnet_ctx)
+    
+        # If the number of requests coming into the preprocess handler is not equal to the batch_size, then we pad the NDArray with 0's.
+        if (self._batch_size - self._num_requests) != 0:
+            padding = mx.nd.zeros((self._batch_size - self._num_requests, c, h, w), self.mxnet_ctx, 'uint8')
+            reqs = mx.nd.concat(reqs, padding, dim=0)
+    
+        return reqs
+```
+**NOTE: The above code handles the case where the `handler` doesn't receive a `batch_size` number of requests. This is because MMS waits for a `max_batch_delay` amount of time to receive 
+`batch_size` number of requests. If the `max_batch_delay` timer time's out before receiving `batch_size` number of requests, MMS frontend bundles what ever requests it received
+and sends it to the handler for processing.**
+
+#### Inference logic
+The inference logic is similar to the inference logic of processing single requests. Since this isn't as interesting, we will skip explaining this in detail. Sample logic is shown below for
+completeness of this document.
+
+```python
+import mxnet as mx
+from collections import namedtuple
+
+class BatchingExample(object):
+...
+    def initialize(self, context):
+        ...
+...
+    def preprocess(self, request):
+        ...
+...
+    def inference(self, model_input):
+        Batch = namedtuple('Batch', ['data'])
```

Comment:
Batch should be lowercase.

Reply:
Oh, any reason why? The code I have here is a working code. I haven't tested with lowercase.

Reply:
local variable naming convention is lowercase with underscores.

Reply:
Ohh, the variable called Batch. Ok, sorry, I thought you meant the tuple value called 'Batch'. You have  a valid point.